test/functorch/test_aotdispatch.py

```diff
@@ -2745,6 +2745,67 @@ def forward(self, x):
         with self.assertRaisesRegex(Exception, "Can't call metadata"):
             make_fx(m, tracing_mode="symbolic", _allow_non_fake_inputs=True)(inp)
 
+    def _compile_and_erase_bases(self, *output_view_indices):
+        # Overrides _base and _view_func tensor attributes, so as to avoid the view-replay
+        # execution path when reconstructing views.
+        class NoViewReplayTensor(torch.Tensor):
+            @property
+            def _base(self):
+                return None
+
+            @property
+            def _view_func(self):
+                return None
+
+        # Wraps the outputs that are views of the FX graph 'g' with NoViewReplayTensor,
+        # since they are the only ones that will get reconstructed.
+        def wrapper(g, *args, **kwargs):
+            outs = g(*args, **kwargs)
+            for i in output_view_indices:
+                outs[i] = NoViewReplayTensor(outs[i])
+            return outs
+
+        return lambda f: aot_function(f, fw_compiler=lambda g, _: partial(wrapper, g))
```

Comment:
@bdhirsh Slightly changed the tests so as not to use `inductor`. Instead, I wrap the output views with a class that overwrites `_base` and `_view_func`, avoiding the view-replay path. Since these outputs are reconstructed from their base tensor, the function will return plain tensors.

Let me know what you think.

Reply:
> avoiding the view-replay path

Hmm i'm not sure I follow. Your new view-replay code doesn't rely on `_base` or `_view_func`, since it tracks the views run during functionalization. The old view-replay path (from autograd) does rely on these attributes, but your new path runs first, so I think i'd expect us to effectively never hit the `_view_func` path unless your path errors (e.g. due to dynamic shapes).

Does any of that sounds wrong?

Reply:
No. You are correct. But, without skipping the view-replay (old) path, this test would pass without this PR. This tests whether we get the same `grad_fn` (`ViewBackward`) class without relying on `_base` and `_view_func`. Did that make sense?

Reply:
ah yep, that's fair!